File: templates/problems/19_acm_io/02_eof_multi_case/typescript/solution.ts
/**
 * Read pairs (a, b) until EOF, return a+b for each pair.
 * Time: O(n), Space: O(n) for output.
 */
export function solveEofMultiCase(inputData: string): string {
    const tokens = inputData.split(/\s+/).filter(t => t.length > 0);
    const outLines: string[] = [];
    for (let i = 0; i + 1 < tokens.length; i += 2) {
        const a = parseInt(tokens[i], 10);
        const b = parseInt(tokens[i + 1], 10);
        outLines.push(String(a + b));
    }
    if (outLines.length === 0) return "";
    return outLines.join("\n") + "\n";
}

File: templates/problems/19_acm_io/02_eof_multi_case/python/solution.py
def solve_eof_multi_case(input_data: str) -> str:
    """Read pairs (a, b) until EOF, return a+b for each pair.

    Time: O(n), Space: O(n) for output.
    """
    tokens = input_data.split()
    out_lines = []
    i = 0
    while i + 1 < len(tokens):
        a = int(tokens[i])
        b = int(tokens[i + 1])
        out_lines.append(str(a + b))
        i += 2
    if not out_lines:
        return ""
    return "\n".join(out_lines) + "\n"
